src/legacy/core_plugins/kibana/public/visualize/np_ready/editor/editor.js

```diff
@@ -313,16 +313,29 @@ function VisualizeAppController(
     }
   );
 
+  const stopAllSyncing = () => {
+    stopStateSync();
+    stopSyncingQueryServiceStateWithUrl();
+    stopSyncingAppFilters();
+  };
+
   // The savedVis is pulled from elasticsearch, but the appState is pulled from the url, with the
   // defaults applied. If the url was from a previous session which included modifications to the
   // appState then they won't be equal.
   if (!_.isEqual(stateContainer.getState().vis, stateDefaults.vis)) {
     try {
       vis.setState(stateContainer.getState().vis);
-    } catch {
-      redirectWhenMissing({
-        'index-pattern-field': '/visualize',
+    } catch (error) {
+      stopAllSyncing();
```

Comment:
Is this necessary? Seems like navigating to `/` will call the scope destroy handler anyway.

Reply:
The `destroy` method won't be invoked, that's why I added this.
I didn't dig inside why this is not happened, but seems it somehow depends on inner angular world (rendering&clearing up the DOM tree)

Reply:
I can confirm, that it's not invoked, so it's a valid way to clean up